src/hooks/useCopyToClipboard.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react'
+
+export interface useCopyToClipboardProps {
+  timeout?: number
+}
+
+export function useCopyToClipboard({ timeout = 2000 }: useCopyToClipboardProps) {
+  const [isCopied, setIsCopied] = useState<boolean>(false)
+
+  const copyToClipboard = (value: string) => {
+    if (typeof window === 'undefined' || !navigator.clipboard?.writeText) {
```

Comment:
Lets also also have the check here so preventing a race condition on `setIsCopied` isnt up to the consumer

```suggestion
    if (isCopied) {
      return
    }

    if (typeof window === 'undefined' || !navigator.clipboard?.writeText) {
```